src/catalog/catalog.cpp

```diff
@@ -952,10 +952,16 @@ optional_ptr<SchemaCatalogEntry> Catalog::GetSchema(CatalogEntryRetriever &retri
                                                     const string &schema_name, OnEntryNotFound if_not_found,
                                                     QueryErrorContext error_context) {
 	auto entries = GetCatalogEntries(retriever, catalog_name, schema_name);
+	auto &context = retriever.GetContext();
 	for (idx_t i = 0; i < entries.size(); i++) {
+		auto entry_name = IsInvalidCatalog(entries[i].catalog) ? GetDefaultCatalog(retriever) : entries[i].catalog;
+		// check whether the catalog is attached before trying to get it
+		if (!context.db->GetDatabaseManager().GetDatabase(context, entry_name)) {
```

Comment:
Can we use `Catalog::GetCatalogEntry` instead, which returns an optional pointer? That way we don't need to do multiple look-ups or need to resolve the entry name ourselves here